File: 06 - Subprogramas/valor_referencia.py
def por_valor(x):
    x = x + 10
    print(f"Dentro de por_valor: {x}")

def por_referencia(lista):
    lista[0] = lista[0] + 10
    print(f"Dentro de por_referencia: {lista[0]}")

a = 5
por_valor(a)
print(f"Após por_valor: {a}")

b = [5]
por_referencia(b)
print(f"Após por_referencia: {b[0]}")

File: 06 - Subprogramas/valor_referencia.c
#include <stdio.h>

void por_valor(int x) {
    x = x + 10;
    printf("Dentro de por_valor: %d\n", x);
}

void por_referencia(int *x) {
    *x = *x + 10;
    printf("Dentro de por_referencia: %d\n", *x);
}

int main() {
    int a = 5;
    
    por_valor(a);
    printf("Após por_valor: %d\n", a);
    
    por_referencia(&a);
    printf("Após por_referencia: %d\n", a);

    return 0;
}
